website_sale_hide_price/models/product_template.py

```diff
@@ -38,3 +38,21 @@ def _get_combination_info(
             }
         )
         return combination_info
+
+    def _search_render_results(self, fetch_fields, mapping, icon, limit):
+        results_data = super()._search_render_results(
+            fetch_fields, mapping, icon, limit
+        )
+        new_result = []
+        for product, data in zip(self, results_data):
+            if product.website_hide_price:
+                data.update(
+                    {
+                        "price": "<span>%s</span>"
+                        % (product.website_hide_price_message or ""),
+                        "list_price": "<span>%s</span>"
+                        % (product.website_hide_price_message or ""),
```

Comment:
This is only used when the discount is shown (like in before/after price), so we don't need to repeat the message, just leaving the key empty.
```suggestion
                        "list_price": ""),
```